test/common/tracing/zipkin/zipkin_tracer_impl_test.cc

```diff
@@ -0,0 +1,315 @@
+#include <chrono>
+#include <memory>
+#include <string>
+
+#include "common/http/header_map_impl.h"
+#include "common/http/headers.h"
+#include "common/http/message_impl.h"
+#include "common/runtime/runtime_impl.h"
+#include "common/runtime/uuid_util.h"
+#include "common/tracing/http_tracer_impl.h"
+#include "common/tracing/zipkin/zipkin_core_constants.h"
+#include "common/tracing/zipkin/zipkin_tracer_impl.h"
+
+#include "test/mocks/http/mocks.h"
+#include "test/mocks/local_info/mocks.h"
+#include "test/mocks/runtime/mocks.h"
+#include "test/mocks/stats/mocks.h"
+#include "test/mocks/thread_local/mocks.h"
+#include "test/mocks/tracing/mocks.h"
+#include "test/mocks/upstream/mocks.h"
+#include "test/test_common/utility.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::_;
+using testing::Invoke;
+using testing::NiceMock;
+using testing::Return;
+using testing::ReturnRef;
+using testing::Test;
+
+namespace Zipkin {
+
+class ZipkinDriverTest : public Test {
+public:
+  void setup(Json::Object& config, bool init_timer) {
+    ON_CALL(cm_, httpAsyncClientForCluster("fake_cluster"))
+        .WillByDefault(ReturnRef(cm_.async_client_));
+
+    if (init_timer) {
+      timer_ = new NiceMock<Event::MockTimer>(&tls_.dispatcher_);
+      EXPECT_CALL(*timer_, enableTimer(std::chrono::milliseconds(5000)));
+    }
+
+    driver_.reset(new ZipkinDriver(config, cm_, stats_, tls_, runtime_, local_info_, random_));
+  }
+
+  void setupValidDriver() {
+    EXPECT_CALL(cm_, get("fake_cluster")).WillRepeatedly(Return(&cm_.thread_local_cluster_));
+    ON_CALL(*cm_.thread_local_cluster_.cluster_.info_, features())
+        .WillByDefault(Return(0)); // No HTTP2 for zipkin upstreams
+
+    std::string valid_config = R"EOF(
+      {
+       "collector_cluster": "fake_cluster",
+       "collector_endpoint": "/api/v1/spans"
+       }
+    )EOF";
+    Json::ObjectPtr loader = Json::Factory::loadFromString(valid_config);
+
+    setup(*loader, true);
+  }
+
+  const std::string operation_name_{"test"};
+  Http::TestHeaderMapImpl request_headers_{
+      {":authority", "api.lyft.com"}, {":path", "/"}, {":method", "GET"}, {"x-request-id", "foo"}};
+  SystemTime start_time_;
+  Http::AccessLog::MockRequestInfo request_info_;
+
+  std::unique_ptr<ZipkinDriver> driver_;
+  NiceMock<Event::MockTimer>* timer_;
+  Stats::IsolatedStoreImpl stats_;
+  NiceMock<Upstream::MockClusterManager> cm_;
+  NiceMock<Runtime::MockLoader> runtime_;
+  NiceMock<ThreadLocal::MockInstance> tls_;
+  NiceMock<LocalInfo::MockLocalInfo> local_info_;
+  Runtime::RandomGeneratorImpl random_;
```

Comment:
(Just search all the test code for non-mock versions to make sure you get them all). Thanks.